sdks/java/extensions/sql/src/test/java/org/apache/beam/sdk/extensions/sql/meta/provider/bigquery/BigQueryWriteIT.java

```diff
@@ -0,0 +1,227 @@
+import org.apache.beam.sdk.extensions.sql.impl.BeamSqlEnv;
+import org.apache.beam.sdk.extensions.sql.impl.schema.BeamPCollectionTable;
+import org.apache.beam.sdk.extensions.sql.meta.provider.BeamSqlTableProvider;
+import org.apache.beam.sdk.extensions.sql.meta.provider.TableProvider;
+import org.apache.beam.sdk.extensions.sql.meta.store.InMemoryMetaStore;
+import org.apache.beam.sdk.io.gcp.bigquery.BigQueryIO;
+import org.apache.beam.sdk.schemas.Schema;
+import org.apache.beam.sdk.testing.PAssert;
+import org.apache.beam.sdk.testing.TestPipeline;
+import org.apache.beam.sdk.transforms.Create;
+import org.apache.beam.sdk.transforms.MapElements;
+import org.apache.beam.sdk.transforms.SimpleFunction;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.Row;
+import org.apache.calcite.sql.SqlExecutableStatement;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.parser.SqlParseException;
+import org.joda.time.Duration;
+import org.joda.time.Instant;
+import org.joda.time.format.DateTimeFormat;
+import org.joda.time.format.DateTimeFormatter;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/**
+ * Integration tests form writing to BigQuery with Beam SQL.
+ */
+@RunWith(JUnit4.class)
+public class BigQueryWriteIT implements Serializable {
+
+  private static final DateTimeFormatter DATETIME_FORMAT =
+      DateTimeFormat.forPattern("YYYY_MM_dd_HH_mm_ss_SSS");
+
+  private static final Schema SOURCE_SCHEMA =
+      RowSqlTypes
+          .builder()
+          .withBigIntField("id")
+          .withVarcharField("name")
+          .build();
+
+  private static final Schema RESULT_SCHEMA =
+      RowSqlTypes
+          .builder()
+          .withVarcharField("id")
+          .withVarcharField("name")
+          .build();
+
+  @Rule public transient TestPipeline pipeline = TestPipeline.create();
+  @Rule public transient TestPipeline resultsPipeline = TestPipeline.create();
+
+  @Test
+  public void testInsertValues() throws Exception {
+    String bigQueryLocation = createBQTableName("testInsertValues");
+    BeamSqlEnv sqlEnv = newSqlEnv(new BigQueryTableProvider());
+
+    String createTableStatement =
+        "CREATE TABLE ORDERS( \n"
+        + "   id BIGINT, \n"
+        + "   name VARCHAR ) \n"
+        + "TYPE 'bigquery' \n"
+        + "LOCATION '" + bigQueryLocation + "'";
+    executeCreateTable(sqlEnv, createTableStatement);
+
+    executeInsert(pipeline, sqlEnv, "INSERT INTO ORDERS VALUES (1, 'foo')");
+
+    pipeline.run().waitUntilFinish(Duration.standardMinutes(2));
+
+    PAssert
+        .that(allRowsFromBQ(resultsPipeline, bigQueryLocation))
+        .containsInAnyOrder(row(RESULT_SCHEMA, "1", "foo"));
+    resultsPipeline.run().waitUntilFinish(Duration.standardMinutes(2));
+  }
+
+  @Test
+  public void testInsertSelect() throws Exception {
+    String bigQueryLocation = createBQTableName("testInsertSelect");
```

Comment:
Does this leak a table on each run? Perhaps a `@Rule` or anyhow some cleanup.

Reply:
Implemented `TestBigQuery` rule.

My first approach was to rely on dataset policy which would expire the table after a day, mainly because we don't seem to have simple wrappers for BigQuery, and writing `TestBigQuery` proved it not as simple as pubsub.

Current version of `TestBigQuery` is limited for the same reason. For example, we don't have a simple way to convert a BigQueryIO TableRows to Beam Rows with original schema because we don't have access to the same schema in TableRows, we either get Avros with their own schema, or JSON rows with strings. I guess that's from the BQ->GCS export, as I couldn't find a simple way to call something like BQ.queryAndGetRows()